staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward_test.go

```diff
@@ -67,7 +69,35 @@ func testPortForward(t *testing.T, flags map[string]string, args []string) {
 			podPath: "/api/" + version + "/namespaces/test/pods/foo",
 			pfPath:  "/api/" + version + "/namespaces/test/pods/foo/portforward",
 			pod:     execPod(),
-			pfErr:   true,
+			hasErr:  true,
+			pfErr:   "pf error",
+		},
+		{
+			name:    "pod not ready",
+			podPath: "/api/" + version + "/namespaces/test/pods/foo",
+			pfPath:  "/api/" + version + "/namespaces/test/pods/foo/portforward",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test", ResourceVersion: "10"},
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyAlways,
+					DNSPolicy:     corev1.DNSClusterFirst,
+					Containers: []corev1.Container{
+						{
+							Name: "bar",
+						},
+					},
+				},
+				Status: corev1.PodStatus{
```

Comment:
What about a test case where the `PodReady` condition is found, but the status is not `ConditionTrue`?  Is that a case that needs to be checked?